src/main/java/org/kairosdb/datastore/cassandra/DataCache.java

```diff
@@ -68,49 +66,63 @@ public DataCache(int cacheSize)
 	}
 
 	/**
-	 returns null if item was not in cache.  If the return is not null the item
+	 returns null if item is not in cache.  If the return is not null the item
 	 from the cache is returned.
 
 	 @param cacheData
 	 @return
 	 */
-	public T cacheItem(T cacheData)
+	public synchronized T get(T cacheData)
 	{
-		LinkItem<T> mappedItem = null;
+		final LinkItem<T> mappedItem = getMappedItem(cacheData);
+
+		pruneCache();
+
+		return (mappedItem == null ? null : mappedItem.m_data);
+	}
 
-		synchronized (m_lock)
+	private synchronized LinkItem<T> getMappedItem(T cacheData) {
```

Comment:
nit: could we rename this to something to indicate it performs a frequency update as well?